### PR TITLE
Add Registry client integration tests

### PR DESCRIPTION
**Pull request by gastaldi**

### CODE REVIEW

**Review by aloubyansky — APPROVED** [review]

File: .github/native-tests.json [hunk 1]
```diff
@@ -137,6 +137,12 @@
             "timeout": 20,
             "test-modules": "hibernate-validator",
             "os-name": "windows-latest"
+        },
+        {
+            "category": "DevTools Integration Tests",
+            "timeout": 70,
```
Comment (famod) [inline .github/native-tests.json:143]: 70 minutes? How long does it take on average?

Comment (gastaldi) [reply .github/native-tests.json:143]: 2-3 minutes, maybe I should lower that a bit :)

**Comment by famod** [comment]
So right now the JVM test execution will take place in the JVM test jobs, not in the Devtools test jobs. Is that what you want?

**Comment by gastaldi** [comment]
> So right now the JVM test execution will take place in the JVM test jobs, not in the Devtools test jobs. Is that what you want?

The Devtools test job created in this PR is meant to run native tests only

**Comment by famod** [comment]
@gastaldi What I mean is that CI has two separate `Devtools Tests` jobs (one for linux and one for windows). Those jobs are only executing `integration-tests/devtools` but not this new module here.
Together with the rest of the CI setup, this is now run in the three `JVM Test` jobs:
```
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ quarkus-integration-test-devtools-registry-client ---
[INFO] Surefire report directory: /home/runner/work/quarkus/quarkus/integration-tests/devtools-registry-client/target/surefire-reports
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.quarkus.registry.RegistryConfigTest
2021-07-12 20:58:55,138 INFO  [io.quarkus] (main) Quarkus 999-SNAPSHOT on JVM started in 2.307s. Listening on: http://localhost:8081
2021-07-12 20:58:55,140 INFO  [io.quarkus] (main) Profile test activated. 
2021-07-12 20:58:55,140 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy, resteasy-jackson, smallrye-context-propagation]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 7.625 s - in io.quarkus.registry.RegistryConfigTest
2021-07-12 20:58:57,258 INFO  [io.quarkus] (main) Quarkus stopped in 0.064s
```

So, my point is that it might be confusing that (some) devtools tests are _not_ run in the dedicated devtools jobs, but in the JVM jobs instead. /cc @gsmet

If you all think that's ok it'll be ok for me too, it's only a minor thing after all.

**Comment by aloubyansky** [comment]
I think it makes sense to add this test to the existing devtools jobs.

**Comment by gastaldi** [comment]
AFAIU the main reason why I didn't add to the `integration-tests/devtools` module is because it does not run or contain native tests, so I decided to not mess with that by putting in anofher module.

**Comment by aloubyansky** [comment]
Could we add it to the existing jobs though?

**Comment by famod** [comment]
Deselection `-pl!...` isn't recursive though, so we'd need to list all devtool modules.
A trick would be to put modules in the devtools IT parent in a profile that can be switched off.
Doing that one level up for all ITs would be rather verbose given the number of modules.

**Comment by famod** [comment]
I can have a look, but not before the weekend I think.

**Pull request merged**